sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -102,15 +107,37 @@ public Result implement(EnumerableRelImplementor implementor, Prefer prefer) {
     }
   }
 
-  private static PipelineResult run(
+  private static @Nullable PipelineResult run(
       PipelineOptions options, BeamRelNode node, DoFn<Row, Void> doFn) {
     Pipeline pipeline = Pipeline.create(options);
     PCollectionTuple.empty(pipeline).apply(node.toPTransform()).apply(ParDo.of(doFn));
     PipelineResult result = pipeline.run();
+
+    if (containsUnboundedPCollection(pipeline)) {
+      return null;
```

Comment:
Nit: Why not return the result anyway?

Reply:
https://github.com/apache/beam/pull/5687/files#diff-4cc8078360c76fc39f49629dd6a630c2R184

Because if I did that I couldn't use null to convey a bit of information and I'd have to wrap this in a new class.

Reply:
But there's `pipelineResult.getState()` which you can check whether it is done yet, can't it be used for the same purpose?

Reply:
nitpicking further, `null` doesn't convey any meaningful information in this context unless you know all the code. Does `null` mean "pipeline is still running but we're not waiting" or "we didn't even run the pipeline because some reasons", or "something is wrong, go crash"?

Reply:
How about just in `count` doing `if (containsUnboundedPCollections(pipeline)) { return 0; }`. And you might want to comment next to that with a JIRA about finding a good way to send a warning / message to the user.

Reply:
pipelineResult.getState doesn't communicate if a pipeline is expected to terminate. It has the same problems as null. I've eliminated the common run function so I will have access to everything I need to make the right decision.

Reply:
Makes sense. Looks good